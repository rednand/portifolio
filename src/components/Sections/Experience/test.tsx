import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Experiences from '.'

describe('Experiences Component', () => {
  it('renders the Experiences component correctly', () => {
    const { container } = render(<Experiences />)

    expect(screen.getByText('ExperiÃªncia')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  // it('should render all experience items from the map', () => {
  //   render(<Experiences />)

  //   mockExperiences.forEach((item) => {
  //     expect(screen.getByText(item.jobTitle)).toBeInTheDocument()
  //     expect(screen.getByText(item.company)).toBeInTheDocument()
  //     expect(
  //       screen.getByText(`${item.startDate} - ${item.endDate}`)
  //     ).toBeInTheDocument()
  //   })
  // })

  it('should render the background color correctly', () => {
    const { container } = render(<Experiences />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#2c2a32' })
  })
})
