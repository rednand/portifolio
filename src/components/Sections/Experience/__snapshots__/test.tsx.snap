// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiences Component renders the Experiences component correctly 1`] = `
.c0 {
  height: 100vh;
  padding: 10vh;
  background: #2c2a32;
  font-family: 'Segoe UI',Roboto,'Helvetica Neue',sans-serif;
  margin: auto;
  scroll-snap-align: start;
}

.c1 {
  font-size: 2.5rem;
  line-height: 120%;
  width: 70%;
  color: #fefefe;
  font-family: 'Segoe UI',Roboto,'Helvetica Neue',sans-serif;
  font-weight: 700;
  margin-bottom: 3%;
}

.c3 {
  width: 80%;
  display: flex;
  justify-content: space-between;
  margin-bottom: 2%;
}

.c4 {
  width: 20%;
  padding-right: 2%;
}

.c6 {
  width: 80%;
  padding-top: 1%;
}

.c7 {
  font-size: 2rem;
  color: #fefefe;
}

.c8 {
  font-size: 1.5rem;
  color: #fefefe;
  padding: 1% 0;
  font-weight: 400;
}

.c5 {
  font-size: 1.2rem;
  color: #fefefe;
  border-top: 10px solid #48d8a4;
  font-weight: 400;
  padding-top: 1%;
}

.c9 {
  font-size: 1.2rem;
  color: #fefefe;
  font-weight: 400;
}

.c9 a {
  color: #48d8a4;
  text-decoration: none;
}

.c9 a:hover {
  text-decoration: underline;
}

@media (max-width: 600px) {
  .c0 {
    padding: 3vh;
  }
}

@media (min-width: 600px) and (max-width: 1000px) {
  .c0 {
    height: 110vh;
  }
}

@media (max-width: 600px) {
  .c1 {
    font-size: 2rem;
    margin: 8% 0;
  }
}

@media (max-width: 600px) {
  .c2 {
    display: flex;
    width: 100%;
    justify-content: space-between;
    overflow: auto;
  }
}

@media (max-width: 600px) {
  .c3 {
    margin-right: 6%;
    width: 100%;
    display: block;
  }
}

@media (max-width: 600px) {
  .c4 {
    width: 100vw;
    padding-right: 0%;
  }
}

@media (max-width: 600px) {
  .c6 {
    width: 100%;
    margin-bottom: 3%;
  }
}

@media (max-width: 600px) {
  .c7 {
    font-size: 1rem;
    margin: 2% 0;
  }
}

@media (max-width: 600px) {
  .c8 {
    font-weight: 600;
    font-size: 1rem;
  }
}

@media (max-width: 600px) {
  .c5 {
    font-size: 1.1rem;
    padding-top: 3%;
  }
}

@media (max-width: 600px) {
  .c9 {
    font-size: 1rem;
    padding-bottom: 3%;
  }
}

<section
  class="c0"
>
  <h3
    class="c1"
  >
    Experiência
  </h3>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <h5
          class="c5"
        >
          Outubro/2021
           - 
          Atual
        </h5>
      </div>
      <div
        class="c6"
      >
        <h3
          class="c7"
        >
          Desenvolvedor Front-End
        </h3>
        <h4
          class="c8"
        >
          Multiplica - Crédito e Investimento
        </h4>
        <p
          class="c9"
        >
          Responsável pela refatoração e manutenção de um sistema de gestão de contas a receber/pagar desenvolvido com arquitetura de Micro front-end utilizando Single-SPA, ReactJS, JavaScript, Material UI, Chakra UI, Redux, entre outras tecnologias. Atuei também no suporte a um sistema robusto de operações financeiras, baseado em ReactJS com Single-SPA, React-Query, Redux, TypeScript e Material UI. Desenvolvi e mantenho um sistema para geração automática de assinaturas no Outlook, utilizando HTML, JavaScript e CSS. Além disso, criei a landing page da empresa de tecnologia Multiplica, disponível em 
          <a
            href="https://beyondbanking.com.br"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://beyondbanking.com.br
          </a>
          .
        </p>
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <h5
          class="c5"
        >
          Maio/2022
           - 
          Fevereiro/2024
        </h5>
      </div>
      <div
        class="c6"
      >
        <h3
          class="c7"
        >
          Desenvolvedor Front-End
        </h3>
        <h4
          class="c8"
        >
          DoarBem - Trabalho Voluntário
        </h4>
        <p
          class="c9"
        >
          Desenvolvi, em conjunto com a squad, a landing page da ONG utilizando ReactJS, Sass, HTML e JavaScript, disponível em 
          <a
            href="https://www.doarbem.com.br"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://www.doarbem.com.br
          </a>
          . Também atuei no desenvolvimento e na manutenção de uma aplicação web organizacional, construída com ReactJS, Sass, Firebase, entre outras tecnologias.
        </p>
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <h5
          class="c5"
        >
          Janeiro/2021
           - 
          Outubro/2021
        </h5>
      </div>
      <div
        class="c6"
      >
        <h3
          class="c7"
        >
          Estagiário em Qualidade de Software
        </h3>
        <h4
          class="c8"
        >
          Kantar Ibope Media
        </h4>
        <p
          class="c9"
        >
          Responsável por oferecer suporte ao time de Qualidade de Software, realizando a execução de casos de teste manuais e automatizados utilizando Python. Experiência com metodologias ágeis, como Kanban e Scrum, além do uso de Azure DevOps, com foco em Test Plans.
        </p>
      </div>
    </div>
  </div>
</section>
`;
